policy/policy.go

```diff
@@ -2,14 +2,90 @@
 // SPDX-License-Identifier: Apache-2.0
 package policy
 
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// PolicyID identifies a specific policy. This is used to retrieve the policy
+// from the store.
+type PolicyID struct {
+	// TenantId  is the ID of the tenant that owns this policy.
+	TenantId string `json:"tenant_id"`
+
+	// Scheme is the name of the scheme with which this policy is associated
+	Scheme string `json:"scheme"`
+
+	// Name is the name of this policy
+	Name string `json:"name"`
+}
+
+// PolicyIDFromKey parses the specified string containing a policy store key
+// into a PolicyID.
+func PolicyIDFromStoreKey(s string) (PolicyID, error) {
+	var pid PolicyID
+
+	parts := strings.Split(s, ":")
+	if len(parts) != 3 {
+		return pid, fmt.Errorf(
+			"bad policy store key %q: want 3 :-separated parts, found %d",
+			s, len(parts),
+		)
+	}
+
+	pid.TenantId = parts[0]
+	pid.Scheme = parts[1]
+	pid.Name = parts[2]
+
+	return pid, pid.Validate()
```

Comment:
From Line 12 to Line 65 should go to file id.go and the file id.go should be renamed to policyID.go